lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -2766,21 +2766,23 @@ LogicalResult ConvertAtenOp<AtenPermuteOp>::matchAndRewrite(
         op,
         "Only ranked tensor types with static shapes are currently supported");
 
-  SmallVector<int64_t> dimListInt;
-  if (!matchPattern(adaptor.getDims(), m_TorchListOfConstantInts(dimListInt)))
+  SmallVector<int64_t> dimListInt64;
+  if (!matchPattern(adaptor.getDims(), m_TorchListOfConstantInts(dimListInt64)))
     return rewriter.notifyMatchFailure(
         op, "Only constant dimensions are currently supported");
+  SmallVector<int32_t> dimListInt32;
```

Comment:
Why not get them as a list of int32 in the first place? Is it necessary to have them as int64 when you're going to cast it anyway?

Reply:
I tried, but m_TorchListOfConstantInts only supports int64_t 👀